src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/UtcDateTimeToLocalStringConverter.shared.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Extensions.Internals;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	/// <summary>
+	/// Converts the incoming value to a <see cref="string"/> local time represenation in the
+	/// datetime format provided.
+	/// </summary>
+	public class UtcDateTimeToLocalStringConverter : ValueConverterExtension, IValueConverter
+	{
+		/// <summary>
+		/// Gets or sets the datetime format value of the provided datetime or datetimeoffset for
+		/// the <see cref="UtcDateTimeToLocalStringConverter"/>. This is a bindable property.
+		/// </summary>
+		public string DateTimeFormat
+		{
+			get => (string)GetValue(DateTimeFormatProperty);
+			set => SetValue(DateTimeFormatProperty, value);
+		}
+
+		/// <summary>
+		/// Backing BindableProperty for the <see cref="DateTimeFormat"/> property.
+		/// </summary>
+		public static readonly BindableProperty DateTimeFormatProperty = BindableProperty.Create(nameof(DateTimeFormat), typeof(string), typeof(UtcDateTimeToLocalStringConverter), defaultValue: "g");
+
+		public object Convert(object? value, Type? targetType, object? parameter, CultureInfo? culture)
+		{
+			if (value is DateTime or DateTimeOffset)
+			{
+				if (!string.IsNullOrEmpty(DateTimeFormat))
+				{
+					if (DateTimeFormat is string dateTimeFormat && !IsValidDateFormat(DateTimeFormat))
```

Comment:
DateTimeFormat is always a string, isn't it?

Reply:
@AndreiMisiukevich Yes, proposing to remove `DateTimeFormat is string dateTimeFormat`? Yes we can remove this and keep the check against a valid DateTimeFormat... for example a date time format of "asd" would be invalid. Some unit tests check for this.